### PR TITLE
Backport database selector flag

### PR DESCRIPTION
**Pull request by chaotixkilla**

### CODE REVIEW

**Comment by chaotixkilla** [comment]
[Backport database selector flag](https://trello.com/c/ZEMWK7tw/1026-backport-database-selector-flag)

**Review by josepfo — APPROVED** [review]

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+LAMASSU_DB=RELEASE
```
Comment (RafaelTaranto) [inline .env]: Can we rename this .env to .env.dev or .env.sample and add .env to gitignore?

File: lib/options-loader.js [hunk 2]
```diff
@@ -2,8 +2,15 @@ const fs = require('fs')
...
 const argv = require('minimist')(process.argv.slice(2))
+const _ = require('lodash/fp')
 
-const STRESS_TEST_DB = 'psql://postgres:postgres123@localhost/lamassu_stress'
+require('dotenv').config()
+
+const dbMapping = psqlConf => ({
+  STRESS_TEST: _.replace('lamassu', 'lamassu_stress', psqlConf),
+  RELEASE: _.replace('lamassu', 'lamassu_release', psqlConf),
+  DEV: _.replace('lamassu', 'lamassu', psqlConf)
```
Comment (RafaelTaranto) [inline lib/options-loader.js]: Maybe we should add a PROD flag that's exactly the same as dev for now.

File: lib/options-loader.js [hunk 3]
```diff
@@ -32,9 +39,7 @@ function load () {
       opts: JSON.parse(fs.readFileSync(globalConfigPath))
     }
 
-    if (argv.testDB) {
-      config.opts.postgresql = STRESS_TEST_DB
-    }
+    config.opts.postgresql = dbMapping(config.opts.postgresql)[process.env.LAMASSU_DB]
```
Comment (RafaelTaranto) [inline lib/options-loader.js]: Let's default to to PROD.

File: lib/options-loader.js [hunk 4]
```diff
@@ -45,9 +50,7 @@ function load () {
         opts: JSON.parse(fs.readFileSync(homeConfigPath))
       }
 
-      if (argv.testDB) {
-        config.opts.postgresql = STRESS_TEST_DB
-      }
+      config.opts.postgresql = dbMapping(config.opts.postgresql)[process.env.LAMASSU_DB]
```
Comment (RafaelTaranto) [inline lib/options-loader.js]: default to PROD as well

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

File: lib/options-loader.js [hunk 5]
```diff
@@ -2,8 +2,16 @@ const fs = require('fs')
 const path = require('path')
 const os = require('os')
 const argv = require('minimist')(process.argv.slice(2))
+const _ = require('lodash/fp')
 
-const STRESS_TEST_DB = 'psql://postgres:postgres123@localhost/lamassu_stress'
+require('dotenv').config({ path: `.env${process.env.NODE_ENV !== 'production' ? `.dev` : ``}` })
```
Comment (RafaelTaranto) [inline lib/options-loader.js]: I don't think we should do this either. The idea here is for everyone to just move the .env over and have it not versioned with whatever config they need. Right now I don't think even prod is using NODE_ENV.

File: lib/options-loader.js [hunk 6]
```diff
@@ -32,9 +40,11 @@ function load () {
       opts: JSON.parse(fs.readFileSync(globalConfigPath))
     }
 
-    if (argv.testDB) {
-      config.opts.postgresql = STRESS_TEST_DB
-    }
+    config.opts.postgresql = dbMapping(config.opts.postgresql)[
+      process.env.LAMASSU_DB === '' || !process.env.LAMASSU_DB
```
Comment (RafaelTaranto) [inline lib/options-loader.js]: Isn't `!process.env.LAMASSU_DB` enough?

File: lib/options-loader.js [hunk 7]
```diff
@@ -45,9 +55,17 @@ function load () {
...
+        process.env.LAMASSU_DB === '' || !process.env.LAMASSU_DB
+        ? 'PROD'
+        : process.env.LAMASSU_DB
+      ])
+
+      config.opts.postgresql = dbMapping(config.opts.postgresql)[
+        process.env.LAMASSU_DB === '' || !process.env.LAMASSU_DB
+        ? 'PROD'
+        : process.env.LAMASSU_DB
+      ]
```
Comment (RafaelTaranto) [inline lib/options-loader.js]: If htis is all inside the same function can't we decided on the env on top and reuse a db variable?

**Review by RafaelTaranto — APPROVED** [review]

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**